tests/testw3emc.f

```diff
@@ -0,0 +1,317 @@
+       PROGRAM TESTW3MC
+C
+C MAIN PROGRAM: TESTW3MC     WRITE AN INDEX FILE
+C   PRGMMR: BOI VUONG        DATE: 2021-01-14
+C
+C ABSTRACT: THIS IS A TEST PROGRAM FOR NCEPLIBS-W3EMC.
+C
+C PROGRAM HISTORY LOG:
+C   2021-01-14 BOI VUONG
+C
+C USAGE: testw3emc gribfile indexfile
+C
+C INPUT FILE:
+C   gribfile     GRIB1 FILE
+C
+C OUTPUT FILE:
+C   indexfile    GRIB1 INDEX FILE
+C
+C EXIT STATES:
+C   COND =   0 - SUCCESSFUL RUN
+C   COND =   1 - GRIB MESSAGE NOT FOUND
+C   COND =   2 - INCORRECT ARGUMENTS
+C   COND =   8 - ERROR ACCESSING FILE
+C
+C ATTRIBUTES:
+C   LANGUAGE: FORTRAN 77
+C
+      PARAMETER (MXSIZE=2000000,MXSIZ3=MXSIZE*3)
+      PARAMETER (MSK1=32000,MSK2=4000)
+      PARAMETER (MBUF=320000*128)
+      PARAMETER (LENHEAD=21)
+      CHARACTER CGB*256,CGI*256,CARG*300
+      CHARACTER CIX1*256,CHEAD(2)*81,CBUF*(MBUF),
+     &          CGRIB*40,CIS*8,CPDS*28,CTITLE*86
+      CHARACTER LABEL*5,MPDS*120,NPDS*63,EPDS*53,KPDS*24
+      INTEGER NARG,IARGC,IARG,CHKIRET,OUTPUT,TOTAL
+      INTEGER DUM,JGDS(200),MPDST(200),JREW,LENHEAD,MI,ITOT
+      INTEGER,DIMENSION(8):: ITIME=(/0,0,0,-500,0,0,0,0/)
+      INTEGER,DIMENSION(28) :: HEXPDS
+      INTEGER IFLD(MXSIZE),IBDSFL(12),IBMAP(MXSIZE),IDAWIP(200)
+      INTEGER KGDS(200),KGDSO(200),KPDST(25)
+      INTEGER NPARM,NBUL,PUNUM,MAPNUM(20),NBITS(20)
+      CHARACTER BULHED*6,CPARM*100,DESC*20,EOML*3
+      CHARACTER * 1 GRIB(MXSIZ3),WMOHDR(LENHEAD)
+      CHARACTER * 200 fileb,filei,fileo
+      CHARACTER KWBX(4),TITLE(132)
+      CHARACTER * 1   PDS(28),PDSL(28),CSEP(80),GDS(400)
+      REAL            FLDI(MXSIZE)
+C
+      LOGICAL         IW3PDS
+C
+      HEXPDS=0
+      DATA  OUTPUT/6/
+2100   FORMAT (' INVENTORY FOR FILE = ',A,/)
+2300   FORMAT(' PRODUCT DEFINITION SECTION (PDS) OF EACH RECORD.',/,
+     &   ' SEE NMC OFFICE NOTE 388 ON GRIB (WMO CODE FM 92-IX EXT.)',/)
+2400   FORMAT (/,' END OF INVENTORY FOR FILE = ',A,', ',
+     &       I5,' GRIB RECORDS.',/,' FILE NUMBER ',
+     &       I2,',',' IT HAS ',I15,' BYTES IN FILE.',/)
+2600   FORMAT (' ERROR EXIT W3FP11 = ',I2,1X,',  RECORD  = ',A4,
+     & ',  ',1X,A63,' , BYTES  = ',I6)
+2800   FORMAT(1X,A5,1X,A,/,' PDS EXT. ',A,/,5X,A25,A18,A2,A12,1X,I9)
+2801   FORMAT(1X,A5,1X,A,/,' PDS EXT. ',A18,/,5X,A25,A18,A2,A12,1X,I9)
+2802   FORMAT(1X,A5,1X,A,/,6X,A25,A18,A2,A12,1X,I0)
+C
+      print *, ''
+      print *,'*** Testing NCEPLIBS-w3emc. ***'
+      print *, ''
+C - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+C  GET ARGUMENTS
```

Comment:
Tests should not need arguments...

Reply:
Do you mean to hard code the filename for test file ?

Reply:
The filename is the name of the test file that you are putting on FTP? 

Yes, it can be hardcoded. It will not change, right?

Reply:
No, The test file will in the unit_test (tests) in w3emc library.  Since, It is a small file and filename will not change.
This test file will be used to test multiple routines in w3emc.